### PR TITLE
Add generic build instructions (Cabal)

### PR DESCRIPTION
**Pull request by romanofski**
This patch adds build instructions to use with cabal.

This is mainly targeting users who don't have access to nix or use a
Fedora distro. Cabal is somewhat universal to Haskell so theoretically
using Cabal should work everywhere. However, these instructions don't
give any help in regards problems users could face like:

* missing system dependencies
* GHC problems
* version mismatches
* etc

Fixes https://github.com/purebred-mua/purebred/issues/352

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,3 +69,33 @@ and `nix-env` to install the application:
...
+
+* libtalloc-devel
+* notmuch-devel
+
+Make sure you have at least one of our supported GHC and Cabal-install >= 2.2
+installed. From a cloned checkout you then go:
+
+```
+# Updates the package list needed for cabal to download dependencies
+$ cabal update
```
Comment (frasertweedale) [inline README.md]: ``should be new-update``, to go with ``new-build``

Comment (romanofski) [reply README.md]: See.. I thought that was one of the commands which didn't change. Will update.

**Comment by hughwilliams94** [comment]
As I mentioned in #352 it might be worth adding a link to that issue to help any future arch users looking to build the project.

**Comment by romanofski** [comment]
I've added an additional patch linking to the issue with the Arch Linux workaround. Hopefully it'll help more people to get started until someone picks up maintainer ship (and I guess we've released our first release).

**Pull request merged**